packages/sdk/src/swap/sdk.ts

```diff
@@ -159,46 +157,70 @@ export class SwapSDK {
     swapStatusRequest: SwapStatusRequest,
     options: RequestOptions = {},
   ): Promise<SwapStatusResponse> {
-    return ApiService.getStatus(this.baseUrl, swapStatusRequest, options);
+    return ApiService.getStatus(
+      this.options.backendUrl,
+      swapStatusRequest,
+      options,
+    );
   }
 
   async executeSwap(
     params: ExecuteSwapParams,
-    txOpts: TransactionOptions = {},
+    txOpts: TransactionOptions & { signer?: Signer } = {},
   ): Promise<TransactionHash> {
     const { srcChain, srcAsset, amount } = params;
 
+    const { signer: optsSigner, ...remainingTxOpts } = txOpts;
```

Comment:
allow to pass signer when calling contract interaction methods. this allows to pass the right signer for the source chain of a swap